### DISCUSSION TITLE
pytest throws stack overflow when I have a test suite with 500 tests

### DISCUSSION THREAD

**Discussion by shlomobauer** [pytest-dev/pytest#7866] [Q&A]
wondering if this is a known issue?  how do I fix if so.
help

**Comment by nicoddemus** [comment]
Hi @shlomobauer, 

pytest itself has over 3k tests and is known to run many more than that, so definitely is not a known issue.

Can you provide any more details, like for example a traceback?

**Reply by The-Compiler** [reply]
FWIW I also have a testsuite with ~8500 tests and don't have any issue - @shlomobauer you'll really need to be a bit more specific :wink:

**Comment by Raphasha27** [comment]
This is usually caused by deep recursion in fixture dependencies or parametrization. pytest has a default recursion limit of 1000, and 500 test cases with fixtures that chain can hit it.

**Quick fix:** Increase the recursion limit before running tests:

```python
# conftest.py
import sys
sys.setrecursionlimit(5000)
```

**Better fix:** Find the recursion source. Add `--setup-show` to see fixture chain depth:

```bash
pytest --setup-show -k "test_name"
```

Common causes:
1. **Class-scoped fixtures requesting function-scoped fixtures** — each test creates a new chain
2. **Parametrized fixtures** with 500 values that each depend on another fixture
3. **Conftest.py inheritance** — if you have nested conftest.py files that import each other

If you're using `@pytest.fixture(params=...)` with 500 params, split into separate test files or use `pytest_generate_tests` instead.
